grub.d/05_zfs_linux.py

```diff
@@ -546,7 +549,7 @@ def __init__(self):
 
         grub_boot_on_zfs = zedenv.lib.be.get_property(
             self.root_dataset, 'org.zedenv.grub:bootonzfs')
-        if grub_boot_on_zfs.lower() == ("1" or "yes"):
+        if grub_boot_on_zfs.lower() == "1" or "yes":
```

Comment:
```suggestion
        if grub_boot_on_zfs.lower() == ("1" or "yes"):
```

This should be reverted back, without the parentheses it's incorrect and essentially evaluates to:

```python
if grub_boot_on_zfs.lower() == "1":
    self.grub_boot_on_zfs = True
if "yes": # Always True
    self.grub_boot_on_zfs = True
```

Reply:
Oh, I see. But your current solution is still not correct. The following evaluates to `False`.

    test = "yes"
    test.lower() == ("1" or "yes")

Reply:
```suggestion
        if (grub_boot_on_zfs.lower() == "1") or (grub_boot_on_zfs.lower() == "yes"):
```

Reply:
You're right, good catch.

Reply:
Should be:

```python
if grub_boot_on_zfs.lower() in ("1", "yes"):
```

Reply:
```suggestion
        if grub_boot_on_zfs.lower() in ("1", "yes"):
```